src/detectors/code_injection/code_injection.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=code-injection@v1.0 defects=1}
+var express = require("express")
+var app = express()
+
+var { spawn } = require("child_process")
+function codeInjectionNoncompliant() {
+    app.get("/:command/:params", (req, res) => {
+        // Noncompliant: user-supplied parameters are evaluated as a script.
+        const subProcess = spawn(req.params.command, [req.params.params])
+
+        subProcess.stdout.on("data", (data) => {
+            res.send(`stdout: ${data}`)
+        })
+
+        subProcess.on("close", (code) => {
+            res.send(`child process exited with code ${code}`)
+        })
+    })
+}
+// {/fact}
+
+
+// {fact rule=code-injection@v1.0 defects=0}
+var express = require("express")
+var app = express()
+
+var { spawn } = require("child_process")
+function codeInjectionCompliant() {
+    app.get("/:command/:params", (req, res) => {
+        const command = req.params.command
+        const arguments = req.params.params
+        // Compliant: user-supplied parameters must be in allow-list to be evaluated.
```

Comment:
Nit: it's not exactly an allow-list, right? (Where is the list?) I know it's very similar, one might argue the "list" is the imaginary list of all words that can be matched by the regex, but that's not an explicit list so it might be confusing to call it that. I would either use an actual allow-list, or adjust the comment.